src/pubsub/node_push.erl

```diff
@@ -112,69 +85,9 @@ do_publish_item(ServerHost, PublishOptions,
 do_publish_item(_ServerHost, _PublishOptions, _Payload) ->
     {error, mongoose_xmpp_errors:bad_request()}.
 
-remove_extra_items(Nidx, MaxItems, ItemIds) ->
-    node_flat:remove_extra_items(Nidx, MaxItems, ItemIds).
-
-delete_item(Nidx, Publisher, PublishModel, ItemId) ->
-    node_flat:delete_item(Nidx, Publisher, PublishModel, ItemId).
-
-purge_node(Nidx, Owner) ->
-    node_flat:purge_node(Nidx, Owner).
-
-get_entity_affiliations(Host, Owner) ->
-    node_flat:get_entity_affiliations(Host, Owner).
-
-
-get_node_affiliations(Nidx) ->
-    node_flat:get_node_affiliations(Nidx).
-
-get_affiliation(Nidx, Owner) ->
-    node_flat:get_affiliation(Nidx, Owner).
-
-set_affiliation(Nidx, Owner, Affiliation) ->
-    node_flat:set_affiliation(Nidx, Owner, Affiliation).
-
-get_entity_subscriptions(Host, Owner) ->
-    node_flat:get_entity_subscriptions(Host, Owner).
-
-get_node_subscriptions(Nidx) ->
-    node_flat:get_node_subscriptions(Nidx).
-
-get_subscriptions(Nidx, Owner) ->
-    node_flat:get_subscriptions(Nidx, Owner).
-
-set_subscriptions(Nidx, Owner, Subscription, SubId) ->
-    node_flat:set_subscriptions(Nidx, Owner, Subscription, SubId).
-
-get_pending_nodes(Host, Owner) ->
-    node_flat:get_pending_nodes(Host, Owner).
-
-get_items(Nidx, From, RSM) ->
-    node_flat:get_items(Nidx, From, RSM).
-
-get_items(Nidx, JID, AccessModel, PresenceSubscription, RosterGroup, SubId, RSM) ->
-    node_flat:get_items(Nidx, JID, AccessModel,
-                        PresenceSubscription, RosterGroup, SubId, RSM).
-
-get_item(Nidx, ItemId) ->
-    node_flat:get_item(Nidx, ItemId).
-
-get_item(Nidx, ItemId, JID, AccessModel, PresenceSubscription, RosterGroup, SubId) ->
-    node_flat:get_item(Nidx, ItemId, JID, AccessModel,
-                       PresenceSubscription, RosterGroup, SubId).
-
-set_item(Item) ->
-    node_flat:set_item(Item).
-
-get_item_name(Host, Node, Id) ->
-    node_flat:get_item_name(Host, Node, Id).
-
 node_to_path(Node) ->
```

Comment:
Perhaps may be removed as well?

Reply:
No possible, it's called directly from `gen_pubsub_node`, `gen_pubsub_node` is called from
https://github.com/esl/MongooseIM/blob/c55763c28ef1376e7d778ac03d0f778e3a43bb84/src/pubsub/nodetree_tree.erl#L137